### ISSUE TITLE
checkHTTP should sniff out CMS/app intercepting requests

### ISSUE THREAD

**Issue by alexzorin** [alexzorin/letsdebug#7]
It is common for people to send all requests to WordPress `index.php` or to `proxy_pass` every request to their Node app.

`checkHTTP` should return specific `Problems` when it thinks that is happening and provide concrete recommendations/links on how to configure your webserver or `.htaccess` to handle `/.well-known/acme-challenge` properly.

**Comment by webprofusion-chrisc** [comment]
I'm looking at the API just now and noticed the http-01 check queries ./well-known/acme-challenge/letsdebug-test - is there any expectation that the server might respond with an actual file result there or does a 404 count as success for this test? 

We do a similar test in Certify (windows) which creates a configcheck file in the same folder, and an http 200 result counts as a pass. I'm thinking if we also created a letsdebug-test file that might be useful as well.

**Comment by alexzorin** [comment]
Hi Christopher,

The server response is largely disregarded except to detect a few things like problems with redirects (loops, bad locations).

It's possible that an enhancement to the API could serve your idea, for the client to nominate its own random response:

```json
POST /
{"method":"http-01","domain":"example.org","expect-http-response": "<nonce>"}
```

which would then cause the test to require the correct answer to be present at each endpoint to get a pass.

If you think that would be valuable for your purposes I'd be happy to implement it.

**Comment by webprofusion-chrisc** [comment]
Hmm, yes that would be pretty useful! Perhaps the client could also specify the filename (so it can optionally be random as well, but definitely under .well-known/acme-challenge/ path)

I'm considering using the API as a fallback when our own configuration checks have failed and they want a second opinion :) - users will quite often we adamant that they're site is working fine (when it's only working for them).

Do you have a single server hosting these tests or is it something that scales? If so, I imagine that would start to cost a fair bit if lot of people use it?

**Comment by alexzorin** [comment]
> Do you have a single server hosting these tests or is it something that scales? If so, I imagine that would start to cost a fair bit if lots of people use it?

Nah, costs are fixed and it's built to deal with pressure (realistically the first limit it's likely to hit are artificially imposed ones). I encourage you to send as many tests as you want. Just LMK if you submit tests from a single IP as there's an IP-based rate limit (HTTP 429).

Anyway, I have added the requested functionality. It's in the README.

    curl --data '{"method":"http-01","domain":"whirlpool.net.au","options":{"http_request_path":"letsdebug-test-custom-path/haha","http_expect_response":"needs to match"}}' -H 'content-type: application/json' https://letsdebug.net

https://letsdebug.net/whirlpool.net.au/2570

I would really love to see an ACME client send traffic to this service, even as a fallback. Let me know how you go.
